pennylane/labs/tests/conftest.py

```diff
@@ -0,0 +1,201 @@
+            # exit the tests if the device cannot be created
+            pytest.exit(
+                f"Device {dev_name} cannot be created. To run the device tests on an external device, the "
+                f"plugin and all of its dependencies must be installed."
+            )
+
+        return dev
+
+    return _device
+
+
+def pytest_runtest_setup(item):
+    """Skip tests marked as broken."""
+
+    # skip tests marked as broken
+    for mark in item.iter_markers(name="broken"):
+        if mark.args:
+            pytest.skip(f"Broken test skipped: {mark.args}")
+        else:
+            pytest.skip("Test skipped as corresponding code base is currently broken!")
+
+
+# ============================
+# These functions are required to define the device name to run the tests for
+
+
+def _convert_to_int_or_float(value):
+    try:
+        return int(value)
+    except ValueError:
+        try:
+            return float(value)
+        except ValueError:
+            return value
+
+
+class StoreDictKeyPair(argparse.Action):
+    """Argparse action for storing key-value pairs as a dictionary.
+
+    For example, calling a CLI program with ``--mydict v1=k1 v2=5``:
+
+    >>> parser.add_argument("--mydict", dest="my_dict", action=StoreDictKeyPair, nargs="+")
+    >>> args = parser.parse()
+    >>> args.my_dict
+    {"v1": "k1", "v2": "5"}
+
+    Note that strings will be converted to ints and floats if possible.
+
+    """
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        self._nargs = nargs
+        super().__init__(option_strings, dest, nargs=nargs, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        my_dict = {}
+        for kv in values:
+            k, v = kv.split("=")
+            my_dict[k] = _convert_to_int_or_float(v)
+        setattr(namespace, self.dest, my_dict)
+
+
+def pytest_runtest_makereport(item, call):
+    """Post-processing test reports to exclude those known to be failing."""
+    tr = orig_pytest_runtest_makereport(item, call)
+
+    if "skip_unsupported" in item.keywords and item.config.option.skip_ops:
+        if call.excinfo is not None:
+            # Exclude failing test cases for unsupported operations/observables
+            # and those using not implemented features
+            if (
+                call.excinfo.type == qml.DeviceError
+                and "supported" in str(call.excinfo.value)
+                or call.excinfo.type == NotImplementedError
+            ):
+                tr.wasxfail = "reason:" + str(call.excinfo.value)
+                tr.outcome = "skipped"
+
+    return tr
+
+
+@pytest.fixture(scope="class")
+def import_labs():
+    """
+    Fixture for a scoped import of qml.labs
+    To be used instead of direct imports allowing `import pennylane as qml` to give access to `qml.labs` in the subsequent tests.
+    """
+    import pennylane.labs

```

Comment:
Address code factor: 

```suggestion
    import pennylane.labs  # pylint: disable=import-outside-toplevel, unused-import

```